src/app.rs

```diff
@@ -2443,8 +2444,28 @@ pub fn start(data_sources: Vec<Box<dyn DeferredDataSource>>) {
     env_logger::try_init().unwrap_or(()); // Log to stderr (if you run with `RUST_LOG=debug`).
 
     let native_options = eframe::NativeOptions::default();
+
+    let mut paths = BTreeSet::new();
+    let arguments: Vec<String> = std::env::args().collect();
+    for argument in &arguments[1..] {
+        if !argument.starts_with("--") {
+            let path = Path::new(&argument)
+                .parent()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .to_string();
```

Comment:
All of this makes sense, but also seems like a lot. If there is something better to do Rust-wise, please lmk.